app/models/project.rb

```diff
@@ -145,29 +145,51 @@ def add_observation(obs)
     imgs.each { |img| images.push(img) }
   end
 
+  # Nimmo note: I remember this one was a lot of work to get right.
+  # IIRC, the Arel was definitely more efficient than the AR for this join.
+  # But it's been so long, i honestly don't remember. Viva la small PR!
+
   # Remove observation (and its images) from this project. Saves it.
   def remove_observation(obs)
     return unless observations.include?(obs)
 
     imgs = obs.images.select { |img| img.user_id == obs.user_id }
```

Comment:
This might be better written with ActiveRecord:  obs.images.where(user_id: obs.user_id).  If obs.images has already been loaded, of course, this will cause an unnecessary extra query.  But in the more likely case the relationship has not, then this could save time... in the extremely unlikely case that the observer has attached another user's images to their observation! :)  Okay, maybe you should ignore me...

Reply:
I like the simplicity of that... a lot.  I vote in favor.

Reply:
Didn't work... reverting.